File: super-print-main/super-print-main/app/src/main/java/codes/shahid/rnprinterplugin/types/Printer.kt
package codes.shahid.rnprinterplugin.types

import codes.shahid.rnprinterplugin.types.KitchenInfo

data class Printer(
    val id: String = "",
    val name: String = "",
    val deviceName: String = "",
    val deviceId: String = "",
    val productId: String = "",
    val macAddress: String = "",
    val vendorId: String = "",
    val printerType: String = "usb",
    val printerSize: String = "3-inch",
    val ip: String = "",
    val port: Int = 0,
    val enableReceipts: Boolean = false,
    val enableKOT: Boolean = false,
    val enableBarcodes: Boolean = false,
    val printerWidthMM: String = "72",
    val charsPerLine: String = "44",
    val kitchen: KitchenInfo? = null,
    val kitchenRef: String? = null,
    val kitchenIds: String = "",
    val model: String = "",
    val numberOfPrints: Int = 1,
    val numberOfKotPrints: Int = 1
)

File: tijarah-pos-main/tijarah-pos-main/src/database/printer/printer.ts
import { Column, Entity } from "typeorm";
import { BsonObjectIdTransformer } from "../../utils/bsonObjectIdTransformer";

@Entity("printer")
export abstract class PrinterModel {
  @Column({
    primary: true, // Marks column as primary
    transformer: new BsonObjectIdTransformer(),
    /* Other options... */
  })
  _id: string;

  @Column()
  name: string;

  @Column({ default: "usb" })
  printerType: string;

  @Column()
  device_name: string;

  @Column({ default: "3-inch" })
  printerSize: string;

  @Column()
  device_id: string;

  @Column()
  product_id: string;

  @Column()
  vendor_id: string;

  @Column()
  enableReceipts: boolean;

  @Column({ default: false, nullable: true })
  enableKOT: boolean;

  @Column()
  enableBarcodes: boolean;

  @Column({ default: "72" })
  printerWidthMM: string;

  @Column({ default: "44" })
  charsPerLine: string;

  @Column("simple-json", { nullable: true })
  kitchen?: { name?: string };

  @Column({ nullable: true, default: "" })
  kitchenRef?: string;
}
